src/mdm-data-class.model.ts

```diff
@@ -41,4 +42,23 @@ export type DataClassDetail =
   & Historical;
 
 export type DataClassIndexResponse = MdmIndexResponse<DataClass>;
-export type DataClassDetailResponse = MdmResponse<DataClassDetail>;
\ No newline at end of file
+export type DataClassDetailResponse = MdmResponse<DataClassDetail>;
+
+/**
+ * Represents hierarchy information for a Data Class.
+ */
+export interface DataClassHierarchy {
+  /**
+   * The child data classes under this Data Class.
+   */
+  dataClasses?: DataClassHierarchy[];
+
+  /**
+   * The Data Elements under this Data Class.
+   */
+  dataElements?: DataElementDetail[];
+}
```

Comment:
I find this interface name a little confusing, does something like:

```
export interface DataClassNode {      
   childDataClasses?: DataClassNode[];        
   childDataElements?: DataElementDetail[];        
}
```
seem any more readable to you?

Reply:
I have renamed the interface to `DataModelNode`, I agree that this sounds like a better name. I have not changed the properties of the type though because these are the names returned by Mauro - the JSON is deserialised to these property names.